### ISSUE TITLE
Incomplete tree shaking (cjs?)

### ISSUE THREAD

**Issue by Benjamin-Dobell** [rollup/rollup#898]
## Version
"rollup version 0.34.13"

## Details

I have an ES6 module ("dom") that looks like this:

```

// ... code ...

var isDescendant = function(parent, child) {
    // ... code ...
};

// ... code ...

export {
    addEventListener,
    getElementContentWidth,
    getElementContentHeight,
    removeAllChildren,
    isDescendant,
    loadCss
};
```

The tree shaking is detecting that `isDescendant` is not called, and hence:

```
var isDescendant = function(parent, child) {
    // ... code ...
};
```

is completely removed from my target CJS bundle.

However, the "transpiled" export referring to `isDescendent` is left in place:

```
var dom = Object.freeze({
    addEventListener: addEventListener,
    getElementContentWidth: getElementContentWidth,
    getElementContentHeight: getElementContentHeight,
    removeAllChildren: removeAllChildren,
    isDescendant: isDescendant,
    loadCss: loadCss
});
```

Which results in "ReferenceError: isDescendant is not defined".

**Comment by TrySound** [comment]
@Benjamin-Dobell Like this?
http://rollupjs.org/?shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJtb2R1bGVJZCUyMiUzQSUyMiUyMiU3RCUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0IlMjBhZGRFdmVudExpc3RlbmVyJTIwJTdEJTIwZnJvbSUyMCcuJTJGbW9kdWxlXzEuanMnJTNCJTVDbiU1Q25hZGRFdmVudExpc3RlbmVyKCklM0IlNUNuJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1vZHVsZV8xLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiUyRiUyMC4uLiUyMGNvZGUlMjAuLi4lNUNuJTVDbnZhciUyMGlzRGVzY2VuZGFudCUyMCUzRCUyMGZ1bmN0aW9uKHBhcmVudCUyQyUyMGNoaWxkKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMkYlMkYlMjAuLi4lMjBjb2RlJTIwLi4uJTVDbiU3RCUzQiU1Q24lNUNuJTJGJTJGJTIwLi4uJTIwY29kZSUyMC4uLiU1Q24lNUNuZXhwb3J0JTIwJTdCJTVDbiUyMCUyMCUyMCUyMGFkZEV2ZW50TGlzdGVuZXIlMkMlNUNuJTIwJTIwJTIwJTIwZ2V0RWxlbWVudENvbnRlbnRXaWR0aCUyQyU1Q24lMjAlMjAlMjAlMjBnZXRFbGVtZW50Q29udGVudEhlaWdodCUyQyU1Q24lMjAlMjAlMjAlMjByZW1vdmVBbGxDaGlsZHJlbiUyQyU1Q24lMjAlMjAlMjAlMjBpc0Rlc2NlbmRhbnQlMkMlNUNuJTIwJTIwJTIwJTIwbG9hZENzcyU1Q24lN0QlM0IlMjIlN0QlNUQlN0Q=

**Comment by Benjamin-Dobell** [comment]
@TrySound I've worked out a minimum reproduction case. The indirection (namespace-like behaviour) seems to cause the issue:

http://rollupjs.org/?shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJtb2R1bGVJZCUyMiUzQSUyMiUyMiU3RCUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0IlMjBpbmRpcmVjdGlvbiUyMCU3RCUyMGZyb20lMjAnLiUyRmluZGlyZWN0aW9uLmpzJyUzQiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJ1bnVzZWQuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIydmFyJTIwdW51c2VkRnVuYyUyMCUzRCUyMGZ1bmN0aW9uKG5vZGUpJTIwJTdCJTVDbiU3RCUzQiU1Q24lNUNuZXhwb3J0JTIwJTdCJTVDbiUyMCUyMCUyMCUyMHVudXNlZEZ1bmMlNUNuJTdEJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmluZGlyZWN0aW9uLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHVudXNlZCUyMGZyb20lMjAnLiUyRnVudXNlZC5qcyclM0IlNUNuJTVDbnZhciUyMGluZGlyZWN0aW9uJTIwJTNEJTIwJTdCJTVDbiU1Q3R1bnVzZWQlM0ElMjB1bnVzZWQlNUNuJTdEJTNCJTVDbiU1Q25leHBvcnQlMjAlN0IlMjBpbmRpcmVjdGlvbiUyMCU3RCUzQiUyMiU3RCU1RCU3RA==

**Comment by Rich-Harris** [comment]
Thanks, I've added that reproduction as a test case in #902, where it passes (rather than including the `unusedFunc` declaration, it knows that it doesn't need to reify the `unused` namespace in the first place – in other words, *all* the code disappears which is a better result)

**Comment by Rich-Harris** [comment]
Fixed in 0.35
